### PR TITLE
FE-216: fix - make view patches async again

### PR DESCRIPTION
**Pull request by palashkaria**
### Scope & Purpose

This uses the jobs API to make view updates async again (which got broken in https://github.com/arangodb/arangodb/pull/15385)

- [x] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] Manually tested
- [x] :book: CHANGELOG entry made

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/FE-216

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const result = await route.put(`/view/${encodedOldViewName}/rename`, {
+    name: view.name.normalize()
+  });
+
+  if (result.body.error) {
+    window.arangoHelper.arangoError(
+      "Failure",
+      `Got unexpected server response: ${result.body.errorMessage}`
+    );
+    error = true;
```
Comment (nikhil-varma) [inline js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx:53]: Out of curiosity, would returning the error itself make it more useful than just a `boolean` ? Maybe we could consider returning the error?

Comment (nikhil-varma) [reply js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx:53]: Something like this (_this is something that I recently started following_)

```
const {response, error} = await someAPICall();

if(error) {
 // TODO
}

```

**Review by nikhil-varma — APPROVED** [review]
Leaving a suggestion for future. Doesn't need it to be now. Thought it could be a useful pattern and if agreed upon, can be followed app-wide 🙂 
Otherwise all OK.

File: js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx (hunk 1 above)
Comment (palashkaria) [reply js/apps/system/_admin/aardvark/APP/react/src/views/views/arangoSearchView/usePatchArangoSearchView.tsx:53]: we are using this boolean to stop the next API call (after rename). The error is not required anywhere as we are already showing it in a notification 

Also, this was existing code in search alias view, I just moved the code around; Can implement your suggestion across both views when we consolidate it! Will keep track

File: js/apps/system/_admin/aardvark/APP/react/src/components/hooks/useJobSync.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            .then(data => readJob(data, job.id))
+            .catch(error => onJobError(error, job.id));
+        }
+      });
+    }
+  };
+  useEffect(() => {
+    window.arangoHelper.getAardvarkJobs(checkState);
+
+    let interval = window.setInterval(() => {
```
Comment (pluma) [inline js/apps/system/_admin/aardvark/APP/react/src/components/hooks/useJobSync.ts:56]: @palashkaria consider https://overreacted.io/making-setinterval-declarative-with-react-hooks/

**Review by pluma — APPROVED** [review]
LGTM but see the caveat about setInterval.

**Comment by KVS85** [comment]
- [ ] https://jenkins01.arangodb.biz/job/arangodb-ANY-jslint/28352/

**Pull request merged**